airflow/models/taskinstance.py

```diff
@@ -139,6 +139,13 @@
 PAST_DEPENDS_MET = "past_depends_met"
 
 
+class ArgDeferred:
+    """sentinel."""
+
+
+DEFERRED = ArgDeferred()
```

Comment:
The naming feels a bit weird. Perhaps we could make those functions return `RunResult | None` where `RunResult` is an enum. It would currently has only one member `deferred` but we could potentially use the mechanism for other things.

Reply:
I like it

Reply:
(your suggestion)